monitoring/default.nix

```diff
@@ -150,82 +189,39 @@ in {
+                module = singleton proto;
+              }
+            ))) ++
+            [(mkBlackboxConfig
+              {
+                inherit machine;
+                proto = "https";
+                hostname = hostName name machine;
+                targets = (filter (n: n != "_" && n != "localhost")
+                            nginxSSLVhosts ++ blackboxCfg.staticBlackboxHttpsTargets);
                 module = [ "https_2xx" ];
-              };
-              static_configs = [
-                {
-                  targets = filter (n: n != "_" && n != "localhost")
-                              nginxSSLVhosts ++ blackboxCfg.staticBlackboxHttpsTargets;
-                  labels = { source = host; };
-                }
-              ];
-              relabel_configs = [
-                { source_labels = [ "__address__" ];
-                  regex = "(.*)(:80)?";
-                  target_label = "__param_target";
-                  replacement = ''''${1}'';
-                }
-                { source_labels = [ "__param_target" ];
-                  regex = "(.*)";
-                  target_label = "instance";
-                  replacement = ''''${1}'';
-                }
-                {
-                  source_labels = [];
-                  regex = ".*";
-                  target_label = "__address__";
-                  replacement = "127.0.0.1:9115";
-                }
-              ];
-            }
-            (flatten (flip map (
-              [ "icmp_v4" "tcp_v4" ] ++ optionals blackboxCfg.blackboxCheckIP6 [ "icmp_v6" "tcp_v6" ]
-            ) (name: let
-              suffix = optionalString (elem name [ "tcp_v4" "tcp_v6" ]) ":22";
-            in {
-              job_name = "blackbox_${name}_${host}";
-              scrape_interval = "60s";
-              metrics_path = "/probe";
-              params = {
-                module = [ name ];
-              };
-              static_configs = [
-                {
-                  targets = (flip map (hostNames allHosts) (h: h + suffix)) ++
-                    (optionals (elem name [ "tcp_v4" "tcp_v6" ]) blackboxCfg.staticBlackboxTcpTargets) ++
-                    (optionals (elem name [ "icmp_v4" "icmp_v6" ]) blackboxCfg.staticBlackboxIcmpTargets);
-                  labels = { source = host; };
-                }
-              ];
-              relabel_configs = [
-                { source_labels = [ "__address__" ];
-                  regex = "(.*)(:80)?";
-                  target_label = "__param_target";
-                  replacement = ''''${1}'';
-                }
-                { source_labels = [ "__param_target" ];
-                  regex = "(.*)";
-                  target_label = "instance";
-                  replacement = ''''${1}'';
-                }
-                {
-                  source_labels = [];
-                  regex = ".*";
-                  target_label = "__address__";
-                  replacement = "127.0.0.1:9115";
```

Comment:
Alarm! All prometheus instances always queried the local blackbox exporter! \o/

Reply:
According to the code the idea seemed to be to query the blackbox exporters of the other prometheus instances. The metrics were tagged that way but instead the local blackbox exporter was always queried. Now we explicitely use the local blackbox exporter but not three times… ;)